pkg/controller/daemon/daemoncontroller.go

```diff
@@ -510,6 +527,10 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 	if err != nil {
 		return fmt.Errorf("couldn't get node to daemon pod mapping for daemon set %q: %v", ds.Name, err)
 	}
+	// Check DeletionTimestamp again because it may have been refreshed by getNodesToDaemonPods().
+	if ds.DeletionTimestamp != nil {
```

Comment:
@enisoc yes, I believe modeling this after what we do in the deployment controller is cleaner and easier to reason about:
- GET object from cache
- LIST all related objects
- Abort if parent is deleted. (+status update)
- Apply the business logic (scale up/down) (+status update)

Not sure if such a refactoring makes sense in this PR.